--- a/ldso/ldso/ldso.c
+++ b/ldso/ldso/ldso.c
@@ -112,8 +112,8 @@
 static unsigned int nlist; /* # items in init_fini_list */
 #ifdef __FDPIC__
 /* We need to take the address of _start instead of its FUNCDESC:
-   declare it as void* to control the relocation emitted.  */
-extern void *_start;
+   declare it as an array to use its address without a data load.  */
+extern char _start[];
 #else
 extern void _start(void);
 #endif
