diff --git a/gcc/config/c33/c33.c b/gcc/config/c33/c33.c
--- a/gcc/config/c33/c33.c
+++ b/gcc/config/c33/c33.c
@@ -547,7 +547,6 @@ const_costs (r, c)
      enum rtx_code c;

 {
-  HOST_WIDE_INT high, low;
   switch (c)
     {
     case CONST_INT:
@@ -984,7 +983,7 @@ print_operand (file, x, code)

 	{
 	  const_double_split (x, &high, &low);
-	  fprintf (file, "0x%lx", (long) high);
+	  fprintf (file, "0x%x", (unsigned int) high);
 	}

       else
@@ -1271,8 +1270,9 @@ output_move_double (operands)
       HOST_WIDE_INT high_low[2];
       int i;
       rtx xop[10];
-      high_low[0] = INTVAL (src);
-      high_low[1] = (INTVAL (src) >= 0) ? 0 : -1;
+      high_low[0] = (unsigned HOST_WIDE_INT) INTVAL (src)
+	& ((((unsigned HOST_WIDE_INT) 1) << BITS_PER_WORD) - 1);
+      high_low[1] = INTVAL (src) >> BITS_PER_WORD;
       for (i = 0; i < 2; i++)

 	{
