diff --git a/drivers/iio/adc/Kconfig b/drivers/iio/adc/Kconfig
index c6d3cf68b..ef530c57d 100644
--- a/drivers/iio/adc/Kconfig
+++ b/drivers/iio/adc/Kconfig
@@ -1527,6 +1527,18 @@ config RZT2H_ADC
 	  To compile this driver as a module, choose M here: the
 	  module will be called rzt2h_adc.
 
+config S1C33_ADC
+	tristate "Epson S1C33E07 A/D converter"
+	depends on C33 || COMPILE_TEST
+	depends on COMMON_CLK
+	help
+	  Say yes here to build support for the five-channel 10-bit A/D
+	  converter in the Epson S1C33E07, which the WikiReader uses for
+	  its battery, thermistor and LCD bias voltages.
+
+	  To compile this driver as a module, choose M here: the module
+	  will be called s1c33-adc.
+
 config SC27XX_ADC
 	tristate "Spreadtrum SC27xx series PMICs ADC"
 	depends on MFD_SC27XX_PMIC || COMPILE_TEST
diff --git a/drivers/iio/adc/Makefile b/drivers/iio/adc/Makefile
index 707dd7089..5b8be2e70 100644
--- a/drivers/iio/adc/Makefile
+++ b/drivers/iio/adc/Makefile
@@ -131,6 +131,7 @@ obj-$(CONFIG_ROCKCHIP_SARADC) += rockchip_saradc.o
 obj-$(CONFIG_RZG2L_ADC) += rzg2l_adc.o
 obj-$(CONFIG_RZN1_ADC) += rzn1-adc.o
 obj-$(CONFIG_RZT2H_ADC) += rzt2h_adc.o
+obj-$(CONFIG_S1C33_ADC) += s1c33-adc.o
 obj-$(CONFIG_SC27XX_ADC) += sc27xx_adc.o
 obj-$(CONFIG_SD_ADC_MODULATOR) += sd_adc_modulator.o
 obj-$(CONFIG_SOPHGO_CV1800B_ADC) += sophgo-cv1800b-adc.o
